ecs-cli/integ/cmd_up/up_test.go

```diff
@@ -0,0 +1,149 @@
+// +build integ
+
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Package cmd_up tests the "ecs-cli up" command with various configurations.
+package cmd_up
+
+import (
+	"fmt"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/integ"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"testing"
+	"time"
+)
+
+const (
+	ecsCLIStackNamePrefix = "amazon-ecs-cli-setup-"
+)
+
+// maxNumberOfRetries * sleepDurationInBetweenRetriesInS represents how long we are willing
+// to wait before we fail a test
+const (
+	maxNumberOfRetries               = 10
+	sleepDurationInBetweenRetriesInS = 30
+)
+
+// TestClusterCreation runs the 'ecs-cli up -c <clusterName> --capability-iam --force' command.
+//
+// If there is no CloudFormation stack created, then the test fails.
+func TestClusterCreation(t *testing.T) {
+	// Given
+	cfnClient, ecsClient, clusterName := setup(t)
+	cmd := integ.GetCommand([]string{"up", "-c", clusterName, "--capability-iam", "--force"})
+
+	// When
+	stdout, err := cmd.Output()
+	assert.NoError(t, err, fmt.Sprintf("Error running %v\nStdout: %s", cmd.Args, string(stdout)))
+
+	// Then
+	assertHasCFNStack(t, cfnClient, clusterName)
+	assertHasActiveContainerInstances(t, ecsClient, clusterName)
+
+	// Cleanup the created resources
+	after(cfnClient, ecsClient, clusterName)
+}
+
+// setup initializes all the clients needed by the test.
+func setup(t *testing.T) (cfnClient *cloudformation.CloudFormation, ecsClient *ecs.ECS, clusterName string) {
+	sess, err := session.NewSession()
+	// Fail the test immediately if we won't be able to evaluate it
+	assert.NoError(t, err, "failed to create new session")
+
+	conf := &aws.Config{
+		Region: aws.String(os.Getenv("AWS_DEFAULT_REGION")),
```

Comment:
if you use AWS_REGION instead, this shouldn't be needed. The AWS SDK Go automatically reads it when you create new session. (If I remember correctly in my own experience the SDK Go doesn't seem to respect AWS_DEFAULT_REGION, not sure why).

Reply:
Done